src/main/java/teammates/logic/core/LocalLoggingService.java

```diff
@@ -35,8 +91,84 @@
 
     @Override
     public QueryLogsResults queryLogs(QueryLogsParams queryLogsParams) {
-        // Not supported in dev server
-        return new QueryLogsResults(Collections.emptyList(), null);
+        // Page size is set as a small value to test loading of more logs
+        int pageSize = 10;
+
+        List<GeneralLogEntry> logEntries = new ArrayList<>();
+        for (GeneralLogEntry logEntry : LOCAL_LOG_ENTRIES) {
+            GeneralLogEntry copiedEntry = new GeneralLogEntry(logEntry.getLogName(), logEntry.getSeverity(),
+                    logEntry.getTrace(), logEntry.getInsertId(), logEntry.getResourceIdentifier(),
+                    logEntry.getSourceLocation(), logEntry.getTimestamp());
+            copiedEntry.setDetails(JsonUtils.fromJson(JsonUtils.toJson(logEntry.getDetails()), Map.class));
+            copiedEntry.setMessage(logEntry.getMessage());
+            logEntries.add(copiedEntry);
```

Comment:
This works, but judging by the intent, it's probably better and saving some computation cost if this is done only to the resulting logs (i.e. post-filtering)?

Reply:
Updated to copy after filtering

Reply:
It is updated.